src/Google/GlobalSiteTag.php

```diff
@@ -155,13 +189,164 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
+			$order->get_total(),
 			esc_js( $order->get_currency() ),
 			esc_js( $order->get_id() ),
 		);
+		wp_print_inline_script_tag( $conversion_gtag_info );
+
+		// Get the item info in the order
+		$item_info = [];
+		foreach ( $order->get_items() as $item_id => $item ) {
+			$product_id   = $item->get_product_id();
+			$product_name = $item->get_name();
+			$quantity     = $item->get_quantity();
+			$price        = $item->get_subtotal();
+			$item_info [] = sprintf(
+				'{
+				id: "gla_%s",
+				price: %f,
+				google_business_vertical: "retail",
+				name: "%s",
+				quantity: %d,
+				}',
+				esc_js( $product_id ),
+				$price,
+				esc_js( $product_name ),
+				$quantity,
+			);
+		}
+
+		// Check if this is the first time customer
+		$is_new_customer = $this->is_first_time_customer( $order->get_billing_email() );
+
+		// Track the purchase page
+		$language = $this->wp->get_locale();
+		if ( 'en_US' === $language ) {
+			$language = 'English';
+		}
+		$purchase_page_gtag =
+		sprintf(
+			'gtag("event", "purchase", {
+			ecomm_pagetype: "purchase",
+			send_to: "%s",
+			transaction_id: "%s",
+			currency: "%s",
+			country: "%s",
+			value: %f,
+			new_customer: %s,
+			tax: "%s",
+			shipping: %f,
+			delivery_postal_code: "%s",
+			aw_feed_country: "%s",
+			aw_feed_language: "%s",
+			items: [%s]});',
+			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
+			esc_js( $order->get_id() ),
+			esc_js( $order->get_currency() ),
+			esc_js( $this->wc->get_base_country() ),
+			$order->get_total(),
+			$is_new_customer ? 'true' : 'false',
+			esc_js( $order->get_cart_tax() ),
+			$order->get_total_shipping(),
+			esc_js( $order->get_billing_postcode() ),
+			esc_js( $this->wc->get_base_country() ),
+			esc_js( $language ),
+			join( ',', $item_info ),
+		);
+		wp_print_inline_script_tag( $purchase_page_gtag );
+	}
+
+	/**
+	 * Display the JavaScript code to track the product view page.
+	 */
+	private function display_view_item_event_snippet(): void {
+		// Only display on the product view page.
+		if ( ! is_product() ) {
+			return;
+		}
+		$product        = wc_get_product( get_the_ID() );
+		$view_item_gtag = sprintf(
+			'gtag("event", "view_item", {
+			send_to: "GLA",
+			ecomm_pagetype: "product",
+			value: %f,
+			items:[{
+				id: "gla_%s",
+				price: %f,
+				google_business_vertical: "retail",
+				name: "%s",
+				category: "%s",
+			}]});',
+			wc_get_price_to_display( $product ),
+			esc_js( $product->get_id() ),
+			wc_get_price_to_display( $product ),
+			esc_js( $product->get_name() ),
+			esc_js( join( ' & ', $this->product_helper->get_categories( $product ) ) ),
+		);
+		wp_print_inline_script_tag( $view_item_gtag );
+	}
+
+	/**
+	 * Display the JavaScript code to track all pages.
+	 */
+	private function display_page_view_event_snippet(): void {
+		if ( ! is_cart() ) {
+			wp_print_inline_script_tag(
+				'gtag("event", "page_view", {send_to: "GLA"});'
+			);
+			return;
+		} else {
```

Comment:
Since the return was added in the if block above. We no longer need an else block. All the code below can be moved outside of the if block so we can indent it with one tab less.